src/useScrollSync.ts

```diff
@@ -0,0 +1,91 @@
+import type React from "react";
+import { useEffect } from "react";
+
+interface Options {
+  proportional?: boolean;
+  axis?: "horizontal" | "vertical" | "both";
+}
+
+export const useScrollSync = (
+  refs: React.MutableRefObject<HTMLElement | null>[],
+  { proportional = true, axis = "both" }: Options = {}
+): void => {
+  useEffect(() => {
+    const nodes = refs
+      .map((ref) => ref.current)
+      .filter((n) => n !== null) as HTMLElement[];
+    nodes.map((node) => {
```

Comment:
Should be a `forEach` since you not returning.

Reply:
done